db/users.js

```diff
@@ -0,0 +1,126 @@
+const client = require("./client");
+const bcrypt = require("bcrypt");
+
+// createUser({ username, password })
+// hash the password before storing it to the database
+async function createUser({ username, password }) {
+  try {
+    const hashedPassword = await bcrypt.hash(password, 10);
+
+    const {
+      rows: [user],
+    } = await client.query(
+      `
+              INSERT INTO users(username, email, password) 
+              VALUES($1, $2) 
+              ON CONFLICT (username) DO NOTHING
+              RETURNING *;
+          `,
+      [username, hashedPassword]
+    );
+
+    delete user.password;
+    return user;
+  } catch (error) {
+    throw error;
+  }
+}
+
+// getUser({ username, password })
+// verify the password against the hashed password
+async function getUser({ username, password }) {
+  try {
+    const user = await getUserByUsername(username);
+    const hashedPassword = user.password;
+    const passwordsMatch = await bcrypt.compare(password, hashedPassword);
+
+    if (passwordsMatch) {
+      delete user.password;
+      return user;
+    }
+  } catch (error) {
+    throw error;
+  }
+}
+
+// getUserById(id)
+// select a user using the user's ID. Return the user object.
+// do NOT return the password
+async function getUserById(userId) {
+  try {
+    const {
+      rows: [user],
+    } = await client.query(
+      `
+          SELECT * FROM users
+          WHERE id = $1;
+        `,
+      [userId]
+    );
+
+    if (user === undefined) {
+      return null;
+    }
+
+    delete user.password;
+    return user;
+  } catch (error) {
+    throw error;
+  }
+}
+
+// getUserByUsername(username)
+// select a user using the user's username. Return the user object.
+async function getUserByUsername(username) {
+  try {
+    const {
+      rows: [user],
+    } = await client.query(
+      `
+          SELECT * FROM users
+          WHERE username = $1;
+        `,
+      [username]
+    );
+
+    if (user === undefined) {
+      return null;
+    }
+
+    return user;
+  } catch (error) {
+    throw error;
+  }
+}
+
+// getAdminUser(useId)
+async function getAdminUser(userId) {
+  try {
+    const {
+      rows: [user],
+    } = await client.query(
+      `
+          SELECT * FROM users
+          WHERE "isAdmin" = true;
+        `,
+      [userId]
+    );
+
+    if (user === undefined) {
+      return null;
+    }
+
+    delete user.password;
+    return user;
+  } catch (error) {
+    throw error;
+  }
+}
+
```

Comment:
we might also need an updateAdminUser function so that we can enable or disable admin permissions. we also might need an updateUser function if a user wants to update their name, pwds , email etc. and a deactivateUser function as well? this probably means we might need an "isActive" true or false column on the users table/ thoughts? 

@yeonjupark95  @markangelodabu  @151177